static/app/components/forms/jsonForm.tsx

```diff
@@ -44,9 +44,9 @@ class JsonForm extends Component<Props, State> {
     this.scrollToHash();
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    if (nextProps.location && this.props.location.hash !== nextProps.location.hash) {
-      const hash = nextProps.location.hash;
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.location && this.props.location.hash !== prevProps.location.hash) {
+      const hash = this.props.location.hash;
       this.scrollToHash(hash);
       this.setState({highlighted: hash});
```

Comment:
`getDerivedStateFromProps` on the `setState` and leave the `didUpdate` for `scrollToHash`

Reply:
I think it causes a second render, but it seems like this is how we handle a lot of our props → state.

I'm down to go back and clean this up later. But for now I verified this is working fine